scripts/check_if_pr_is_low_risk.py

```diff
@@ -0,0 +1,292 @@
+            if len(lines) != 2:
+                return 'Only 1 line should change in package.json'
+            if not (
+                    bool(re.match(
+                        r'-  "version": "[0-9]\.[0-9]\.[0-9]",',
+                        lines[0],
+                    )) and bool(re.match(
+                        r'\+  "version": "[0-9]\.[0-9]\.[0-9]",',
+                        lines[1],
+                    ))):
+                return 'package.json changes not low-risk'
+
+        elif old == 'core/templates/pages/about-page/about-page.constants.ts':
+            for line in file_diffs[old]:
+                if not re.match(r'\+    \'[A-Za-z ]+\',', line):
+                    return 'about-page.constants.ts changes not low-risk'
+        else:
+            return 'File %s changed and not low-risk' % old
+    return ''
+
+
+def check_if_pr_is_changelog_pr(pr):
+    """Check if a PR is low-risk by virtue of being a changelog PR.
+
+    To be a low-risk changelog PR, a PR must:
+
+    * Be opened from a branch on the oppia/oppia repository.
+    * Be opened from a branch matching
+      `^update-changelog-for-release-v[0-9.]+$`.
+    * Only change the following files:
+        * AUTHORS
+        * CONTRIBUTORS
+        * CHANGELOG
+        * package.json to update the version number
+        * about-page.constants.ts to add names to the contributors list
+
+    Args:
+        pr: dict. JSON object of PR from GitHub API.
+
+    Returns:
+        str. An empty string if the PR is a changelog PR and low-risk,
+        else a message explaining why the PR is not low-risk.
+    """
+    source_repo = pr['head']['repo']['full_name']
+    if source_repo != 'oppia/oppia':
+        return 'Source repo is not oppia/oppia'
+    source_branch = pr['head']['ref']
+    if not re.match(
+            '^update-changelog-for-release-v[0-9.]+$',
+            source_branch):
+        return 'Source branch does not indicate a changelog PR'
+    diff_files, file_diffs = load_diff(pr['base']['ref'])
+    if not diff_files:
+        return 'Failed to load PR diff'
+    return _check_changelog_pr_diff(diff_files, file_diffs)
+
+
+LOW_RISK_CHECKERS = (
+    ('translatewiki', check_if_pr_is_translation_pr),
+    ('changelog', check_if_pr_is_changelog_pr),
+)
+
+
+def main(tokens=None):
+    """Check if a PR is low-risk."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'pr_url',
+        help='The URL of the pull request.'
+    )
+    args = parser.parse_args(args=tokens)
+    parsed_url = parse_pr_url(args.pr_url)
+    if not parsed_url:
+        raise RuntimeError('Failed to parse PR URL %s' % args.pr_url)
+    owner, repo, number = parsed_url
+    pr = lookup_pr(owner, repo, number)
+    if not pr:
+        raise RuntimeError('Failed to load PR from GitHub API')
+    base_repo_url = pr['base']['repo']['clone_url']
+    common.run_cmd(
+        ['git', 'remote', 'add', UPSTREAM_REMOTE, base_repo_url])
```

Comment:
should we hard code the base URL: https:/github.com/oppia/oppia.git (or the ssh URL)

Reply:
We only query the API once, and it's in the `lookup_pr` function. This function only gets called once. Then we just lookup values from the resulting dictionary. We have to query the API at least once to check the source repo and branch, so I don't think we can reduce API requests any further. Since we have to lookup the PR from the API, we might as well avoid hard-coding URLs, right?